packages/vite/src/node/ssr/ssrStacktrace.ts

```diff
@@ -56,3 +56,20 @@ export function ssrRewriteStacktrace(
     })
     .join('\n')
 }
+
+export function rebindErrorStacktrace(e: Error, stacktrace: string): void {
+  const { configurable, writable } = Object.getOwnPropertyDescriptor(
+    e,
+    'stack'
+  )!
+  if (configurable) {
+    Object.defineProperty(e, 'stack', {
+      value: stacktrace,
+      enumerable: true,
+      configurable: true,
+      writable: true
+    })
+  } else if (writable) {
+    e.stack = stacktrace
+  }
```

Comment:
Should we warn if neither configurable nor writable, so the dependency gets fixed?

Reply:
Hm, emitting the warning properly would require `server` instance wouldn't be accessible inside `rebindErrorStacktrace`, would it be better to:

Pass in a server instance to have `rebindErrorStacktrace` emit the warning?
```ts
function rebindErrorStacktrace(e: Error, stacktrace: string, server: ViteDevServer): void
```

Return a boolean for success and have the respective consumers emit the warning? There probably would also be more context available for more detailed errors.
```ts
function rebindErrorStacktrace(e: Error, stacktrace: string): boolean
```

Reply:
I'm thinking we should deprecate `ssrRewriteStacktrace` in favor of a `SSRError extends Error` subclass whose constructor takes a stack trace string. So for now, I vote to merge this PR as-is.